cscs-checks/libraries/math/trilinos_compile_run.py

```diff
@@ -1,40 +1,42 @@
 import os
-import reframe.utility.sanity as sn
-
-from reframe.core.pipeline import RegressionTest
 
+import reframe as rfm
+import reframe.utility.sanity as sn
 
-class TrilinosTest(RegressionTest):
-    def __init__(self, linkage, **kwargs):
-        super().__init__('trilinos_compile_run_%s' % linkage,
-                         os.path.dirname(__file__), **kwargs)
-
-        self.flags = ' -DHAVE_MPI -DEPETRA_MPI -lparmetis -%s ' % linkage
-        self.prgenv_flags = {
-            'PrgEnv-cray': ' -homp -hstd=c++11 -hmsglevel_4 ',
-            'PrgEnv-gnu': ' -fopenmp -std=c++11 -w -fpermissive ',
-            'PrgEnv-intel': ' -qopenmp -w -std=c++11 ',
-            'PrgEnv-pgi': ' -mp -w '
-        }
-        self.descr = 'Trilinos ' + linkage.capitalize()
-        self.sourcepath = 'example_AmesosFactory_HB.cpp'
-        #self.sourcepath = 'trilinos_compile_run.cpp'
-        self.input_file = os.path.join(self.current_system.resourcesdir,
-                                       'Trilinos', 'trilinos_compile_run.rua')
 
-        self.executable_opts = self.input_file.split()
+@rfm.parameterized_test(['static'], ['dynamic'])
+class TrilinosTest(rfm.RegressionTest):
+    def __init__(self, linkage):
+        super().__init__()
         self.valid_systems = ['daint:gpu', 'daint:mc',
                               'dom:gpu', 'dom:mc']
 
-        # Removed CRAY env in dynamic because of CrayBug/809265
-        if linkage == 'dynamic':
+        # NOTE: PrgEnv-cray in dynamic does not work because of CrayBug/809265
+        # PrgEnv-cray seg faults for cray-trilinos/12.12.1.1
+        if linkage == 'static':
             self.valid_prog_environs = ['PrgEnv-gnu', 'PrgEnv-intel']
-        elif linkage == 'static':
+        else:
             self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-gnu',
                                         'PrgEnv-intel']
 
+        self.build_system = 'SingleSource'
+        self.build_system.ldflags = ['-%s' % linkage]
+        self.build_system.cppflags = ['-DHAVE_MPI', '-DEPETRA_MPI']
+        self.build_system.cxxflags = ['-lparmetis']
+        self.prgenv_flags = {
+            'PrgEnv-cray': ['-homp', '-hstd=c++11', '-hmsglevel_4'],
+            'PrgEnv-gnu': ['-fopenmp', '-std=c++11', '-w', '-fpermissive'],
+            'PrgEnv-intel': ['-qopenmp', '-w', '-std=c++11'],
+            'PrgEnv-pgi': ['-mp', '-w']
+        }
+        self.sourcepath = 'example_AmesosFactory_HB.cpp'
+        input_file = os.path.join(self.current_system.resourcesdir,
+                                  'Trilinos', 'trilinos_compile_run.rua')
+        self.executable_opts = input_file.split()
```

Comment:
This split here is kind of strange, isn't it? I don't think it's needed: `[input_file]` should be fine.